### ISSUE TITLE
Loosen up the requirements

### ISSUE THREAD

**Issue by iver56** [hetznercloud/hcloud-python#53]
Right now, you specify exact versions of dependencies, like this:
```
    "future==0.17.1",
    "python-dateutil==2.7.5",
    "requests==2.20.0"
```

This may be incompatible with existing projects. It would be better to specify _ranges_ of versions that are compatible with the `hcloud` package. For example, `requests` is a very stable package, and `hcloud` will probably work with a wide range of `requests` versions, e.g. `requests>=2.20,<2.23`

**Comment by ssbarnea** [comment]
If These unpining do not happen ASAP, I will be forced to remove the hetznercloud driver from molecule as it it creating installation conflicts:
```
ERROR: hcloud 1.4.0 has requirement requests==2.20.0, but you'll have requests 2.22.0 which is incompatible.
```
The idea of pinning versions in a library is becoming a deal-breaker.

**Comment by LKaemmerling** [comment]
We will look into this on monday.

**Comment by LKaemmerling** [comment]
@ssbarnea @iver56 

We have released v1.4.1 with a fix for this problem.

**Comment by cenkalti** [comment]
I got this error when I use hcloud in my project:
```
ERROR: hcloud 1.8.1 has requirement requests<2.24,>=2.20, but you'll have requests 2.24.0 which is incompatible.
```

Why are you pinning dependencies?

From: https://packaging.python.org/discussions/install-requires-vs-requirements/#install-requires
> It is not considered best practice to use install_requires to pin dependencies to specific versions, or to specify sub-dependencies (i.e. dependencies of your dependencies). This is overly-restrictive, and prevents the user from gaining the benefit of dependency upgrades.

> Lastly, it’s important to understand that install_requires is a listing of “Abstract” requirements, i.e just names and version restrictions that don’t determine where the dependencies will be fulfilled from (i.e. from what index or source). The where (i.e. how they are to be made “Concrete”) is to be determined at install time using pip options.

Required projects use semantic versioning and they can be specified as follows:
```
requirements = [
    "future>=0.17.1,<1",
    "python-dateutil>=2.7.5,<3",
    "requests>=2.20,<3"
]
```

I can send you a PR if you are okay with the idea?

**Comment by ssbarnea** [comment]
I totally agree with that. In fact these invalid requirements made me seriously considering removing support for hertzercloud from molecule bundle, as they create conflicts. If no hotfix is released, I will endup doing this as we validate conflicts.

Semantic makes sense. If there is a know a specific issue with X.Y.Z it is possible to use `!=X.Y.Z` notation and avoid defining a set of requirements that will conflict with almost anything.

**Comment by LD250** [comment]
Hi @cenkalti, 

I think you suggestion makes sense. If Required projects introduce breaking change, they increase the major version. We can loosen up upper bounds. I like your Idea, please send a PR.

**Comment by LD250** [comment]
@cenkalti @ssbarnea 

We released v1.8.2 with a fix for requirements.
